### PR TITLE
Refs #15312 - Add hostgroup create/update tests

### PR DESCRIPTION
**Pull request by akofink**
- Add tests to make sure hammer takes all the options we think it should
  for hostgroups create and update

related: https://github.com/Katello/hammer-cli-katello/pull/401

### CODE REVIEW

File: test/unit/id_resolver_test.rb [hunk 1]
```diff
@@ -190,7 +190,7 @@
             { 'id' => 85, 'name' => 'git::params', 'created_at' => '2015-01-27T07:24:57.306Z', 'updated_at' => '2015-01-27T07:24:57.306Z' }
           ] })
 
-        resolver_run.call.must_equal [70, 27, 85]
+        resolver_run.call.map { |o| o['id'] }.flatten.must_equal [70, 27, 85]
```
Comment (tstrachota) [inline test/unit/id_resolver_test.rb]: This looks like inconsistent change of the interface. Why did you change it to return array of hashes?

File: lib/hammer_cli_foreman/id_resolver.rb [hunk 2]
```diff
@@ -44,9 +48,16 @@ def self.s_name(description, options={})
...
+      :lifecycle_environment => [
+        s_name(_("Lifecycle environment name to search by")),
+        s("organization_id", _("Organization ID"))
+      ],
       :medium =>           [ s_name(_("Medium name")) ],
       :model =>            [ s_name(_("Model name")) ],
-      :organization =>     [ s_name(_("Organization name")) ],
+      :organization =>     [
+        s_name(_("Organization name")),
+        s("label", _("Organization label"))
```
Comment (tstrachota) [inline lib/hammer_cli_foreman/id_resolver.rb]: There's nothing like org labels in Foreman. It's a Katello's extension (https://github.com/Katello/katello/blob/master/app/models/katello/concerns/organization_extensions.rb#L35). Therefore it should be changed in hammer-cli-katello.

File: lib/hammer_cli_foreman/id_resolver.rb [hunk 3]
```diff
@@ -35,6 +35,10 @@ def self.s_name(description, options={})
     SEARCHABLES = {
       :architecture =>     [ s_name(_("Architecture name")) ],
       :compute_resource => [ s_name(_("Compute resource name")) ],
+      :content_view => [
```
Comment (tstrachota) [inline lib/hammer_cli_foreman/id_resolver.rb]: Again, nothing like content views in the Foreman.

File: lib/hammer_cli_foreman/id_resolver.rb [hunk 4]
```diff
@@ -44,9 +48,16 @@ def self.s_name(description, options={})
       # :image =>            [],
       :interface =>        [],
       :location =>         [ s_name(_("Location name")) ],
+      :lifecycle_environment => [
```
Comment (tstrachota) [inline lib/hammer_cli_foreman/id_resolver.rb]: Ditto

**Comment by tstrachota** [comment]
@akofink I appreciate the effort with adding the tests, thanks! Most of the changes should be implemented in hammer-cli-katello as they contain concepts from Katello which are unknown to standalone Foreman.

**Comment by akofink** [comment]
@tstrachota Still want the tests? :vertical_traffic_light:

**Comment by akofink** [comment]
@tstrachota I removed all the Katello-related `disallows ...` tests.

**Comment by jlsherrill** [comment]
[test]

**Comment by tstrachota** [comment]
:+1:  :ant: Thanks @akofink merging!
